### PR TITLE
separate requirements for logger dependencies

### LINKED ISSUE #764: wandb logging does not work - log method called on the wrong object (?)
## 🐛 Bug

When using the WandbLogger and not providing an experiment, I get an `AttributeError: 'Run' object has no attribute 'log'` in line [84 of WandbLogger](https://github.com/PyTorchLightning/pytorch-lightning/blob/deffbaba7ffb16ff57b56fe65f62df761f25fbd6/pytorch_lightning/logging/wandb.py#L84). Instead on `experiment`, I think `log` should be called on `wandb`

#### Code sample
    wandb_logger = WandbLogger(name="name", save_dir="/path/to/folder", offline=False,  project="project")
    trainer = Trainer(logger=wandb_logger)
    trainer.fit(model)

### Environment

 - PyTorch Version (e.g., 1.0): 1.2
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version:9.0
 - GPU models and configuration:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by Aljo-Rovco**
Fixes  #764.
The versions are the current versions of the respective packages.
I used all loggers in offline mode and they all seemed to work. Not sure how exhaustive the test should be.

### CODE REVIEW

**Comment by williamFalcon** [comment]
thanks! why do we need a separate requirements? we already have a different one in tests.

**Comment by Aljo-Rovco** [comment]
ah thanks didn't know about that. The problem is, that we need to specify versions of optional packages (e.g. loggers) somewhere. More as a reference for users than a file for installing requirements.

**Comment by Borda** [comment]
@williamFalcon as a user you do not want to install all testing packages, just those you need to run e.g. special logger... we have a warning message if you ale loading logger for which you do not have a package installed, but it is missing min version...

**Comment by Aljo-Rovco** [comment]
@Borda only test_tube had a minimal version. Copied that, added `-r ../requirements-extra.txt` and removed the loggers from `test/requirements.txt`

**Comment by Borda** [comment]
@Aljo-Rovco it would be great contributions to find out what is the min versions to pass all tests (you can try something like divide an conquer strategy =)

**Comment by Aljo-Rovco** [comment]
will deal with this either today or tomorrow

**Comment by Aljo-Rovco** [comment]
@Borda I assume you meant `tests/test_logging.py`?

**Comment by Aljo-Rovco** [comment]
I increased the amount of data that is used when running the tests (from `0.01` to `0.05`), because otherwise the `log` method in the wandb-logger is not called. And if that method is not called, the tests would run fine with `wandb==0.8.14`. I increased the number for the other loggers as well, just to be on the save side

**Comment by Aljo-Rovco** [comment]
@Borda could you have a look at the failing checks? Not quite sure how to fix them

**Comment by Borda** [comment]
```
lists of files in version control and sdist do not match!
suggested MANIFEST.in rules:
  include *.txt
missing from sdist:
  requirements-extra.txt
```
you need to **include** or ~~exclude~~ the new file in MANIFEST.ini

**Comment by Borda** [comment]
it seems there is an ussie with `wandb` may you have look at it...
```
wandb.run_manager.LaunchError: W&B process failed to launch, see: C:\Users\RUNNER~1\AppData\Local\Temp\wandb-debug.log
```

**Comment by Aljo-Rovco** [comment]
had a look, but not sure how to solve it.
line 1126 in `wandb/run_manager.py` (`stdout_read_file = os.fdopen(stdout_read_fd, 'rb')`) throws `
OSError: [WinError 6] The handle is invalid`. I don't have access to a windows machine, so I cannot recreate/test it.

**Comment by Borda** [comment]
it was working before so let's try to increase the wandb version...

**Comment by Aljo-Rovco** [comment]
I think it was working before because the test didn't test much. It just created a logger:

    def test_wandb_logger(tmpdir):
        """Verify that basic functionality of wandb logger works."""
        tutils.reset_seed()

        wandb_dir = os.path.join(tmpdir, "wandb")
         _ = WandbLogger(save_dir=wandb_dir, anonymous=True) 

All other tests actually trained a model

**Comment by Aljo-Rovco** [comment]
also, isn't the last version used in the CI testing?

File: tests/requirements.txt [hunk 1]
```diff
@@ -5,11 +5,7 @@ pytest>=3.0.5
 pytest-cov
 flake8
 check-manifest
-test-tube>=0.7.5
-mlflow
-comet_ml
-wandb
 sacred
```
Comment (Borda) [inline tests/requirements.txt]: it seems not to be used anywhere, pls remove it...

File: requirements-extra.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+neptune-client>=0.4.4
+comet-ml>=1.0.56
+mlflow>=1.0.0
+test_tube>=0.7.5
+wandb>=0.8.21
```
Comment (Borda) [inline requirements-extra.txt]: add line in .github/workflow/ci-testing.yml
```
python -c "req = open('requirements-extra.txt').read().replace('>', '=') ; open('requirements-extra.txt', 'w').write(req)"
```
just belo the changing `requirements.txt` so we have the min version propoerly tested

**Comment by Borda** [comment]
> also, isn't the last version used in the CI testing?

add line in .github/workflow/ci-testing.yml
```
python -c "req = open('requirements-extra.txt').read().replace('>', '=') ; open('requirements-extra.txt', 'w').write(req)"
```
just belo the changing `requirements.txt` so we have the min version propoerly tested

**Comment by Borda** [comment]
> I think it was working before because the test didn't test much. It just created a logger:
>  ...

I would say you can try to fix it here or remove the new test and open issue where you describe the problem along with your proposed test... I would suggest to to to the second option so we keep the PR simple... sure you can take the new issue and solve it :]

**Comment by Borda** [comment]
@Aljo-Rovco pls create an issue with suggesting the tests you had here for `wandb` so we do not forget it and keep everything well tested :]

**Comment by Aljo-Rovco** [comment]
Sorry, was busy. Will do

**Comment by Aljo-Rovco** [comment]
created a new issue: #906

**Pull request merged**